src/ray/object_manager/pull_manager.cc

```diff
@@ -408,10 +409,14 @@ void PullManager::TryToMakeObjectLocal(const ObjectID &object_id) {
     return;
   }
 
-  // The object waiting for local pull retry; abort.
+  // The object is no longer needed; abort.
   auto it = object_pull_requests_.find(object_id);
-  RAY_CHECK(it != object_pull_requests_.end());
+  if (it != object_pull_requests_.end()) {
```

Comment:
Why the change here? I think we should try to keep the invariant that an object that was in the active pull requests should always be in this map too.

Reply:
The check is failing. I was not able to reproduce the reason why, but removing this add adding a defensive remove above also fixes the problem.

In general, I think we should prefer to write code in a more defensive style (correct even if invariants are violated), if possible.

Reply:
I see. Can you log a warning, though? This case really should not occur so if it does, it's likely a bug.

Reply:
I did a bit more digging and fixed the root cause bug (see other comments), which the defensive erase above was also handling.